### PR TITLE
Update docs with strong prescriptive recommendation on how to handle dynamic form behaviour with custom slot mappings

### PR DESCRIPTION
**Pull request by ancalita**
**Proposed changes**:
- Completes [ATO-442](https://rasahq.atlassian.net/browse/ATO-442)

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [X] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: docs/docs/migration-guide.mdx [hunk 1]
```diff
@@ -1443,9 +1443,17 @@ the mappings will be applied on each user turn regardless of whether a form is a
...
 If you have used `FormValidationAction` to define custom extraction and validation code in which you override the
 `required_slots` method, note that `slots_mapped_in_domain` argument has been replaced by the `domain_slots` argument.
-You will need to make this replacement to continue using your custom code.
+You must make this replacement to continue using your custom code.
+
+If you have been dynamically filling slots not present in the form's `required_slots` defined in the `domain.yml`
+file, note that this behaviour is no longer supported in 3.x. Any dynamic slots with custom mappings, which are set in
+the last user turn, will be filled **only if** they are returned by the `required_slots` method of the custom action
+inheriting from `FormValidationAction`. To maintain the 2.x behaviour, you must now override the `required_slots` method
+of this custom action as per the strong recommendation listed in the [dynamic form documentation](./forms.mdx#dynamic-form-behavior).
```
Comment (l2nguyen) [inline docs/docs/migration-guide.mdx:1456]:
```suggestion
of this custom action as per the strong recommendation listed in the [dynamic form documentation](./forms.mdx#dynamic-form-behavior). 

To extract custom slots that are not defined in any form's `required_slots`, you should now use a global [custom slot mapping](./domain.mdx#custom-slot-mappings) or extend the [ValidationAction class](./action-server/validation-action.mdx#validationaction-class).
```

**Review by l2nguyen — COMMENTED** [review]
Looks good - I made a suggestion on how the user can extract custom global slots in 3.x in the migration guide.

**Review by m-vdb — APPROVED** [review]
💯

**Comment by chandrikas** [comment]
LGTM

**Review by chandrikas — APPROVED** [review]

**Pull request merged**